server/src/logic/sync/domain/use-cases/sync-single-file.use-case.ts

```diff
@@ -95,4 +102,14 @@ export class SyncSingleFile implements SyncSingleFileUseCase {
       }),
     );
   }
+
+  private checkMimetype(url: string): Observable<string> {
+    try {
+      return from(fetch(url, { method: 'HEAD' })).pipe(
+        map((response) => (response.headers.get('content-type') || 'text/html').split(';')[0]),
+      );
+    } catch (err) {
+      return of('text/html');
```

Comment:
After testing some error cases I couldn't get this`catch` to work. Would a `catchError` work?

Reply:
fixed